README.md

```diff
@@ -188,7 +158,119 @@ make run
 
 You should be able to reach the service at: http://0.0.0.0:8080. The port that is used is controlled by an environment variable defined in the `.flaskenv` file which Flask uses to load it's configuration from the environment by default.
 
-Test for webhook
+
+## Behavior-Driven Development (BDD)
+
+Behavior-Driven Development (BDD) is implemented to test the application from the end-user's perspective. It ensures that the service behaves as expected when accessed through its UI. Tests are written in Gherkin syntax and executed using `behave` with Selenium.
+
+---
+
+### BDD Features
+
+The BDD tests validate the following operations:
+1. **Create a Promotion**: Verify that new promotions can be added through the UI.
+2. **Read a Promotion**: Ensure that promotion details can be viewed.
+3. **Update a Promotion**: Test that existing promotions can be updated.
+4. **Delete a Promotion**: Confirm that promotions can be removed.
+5. **List Promotions**: Verify that all promotions are displayed in the UI.
+6. **Query Promotions**: Test the ability to filter promotions by specific criteria.
+7. **Action on Promotions**: Validate custom actions defined for promotions.
+
+---
+
+### Running BDD Tests
+
+To execute the BDD tests:
+
+   ```bash
+   behave
+   ```
+
+## Kubernetes Deployment
+
+This microservice is deployed to a Kubernetes cluster using the manifests provided in the `k8s/` directory.
+
+### Local Development Setup
+
+To deploy the application locally, follow these steps:
+
+1. **Create a Kubernetes Cluster**:
+   - Use `K3s` or `Minikube` to set up a local cluster.
+   - Run the following command to create the cluster:
+     ```bash
+     make cluster
+     ```
+
+2. **Deploy Application and Database**:
+   - Apply the Kubernetes manifests:
+     ```bash
+     kubectl apply -f k8s/
+     ```
+
+3. **Verify Deployment**:
+   - Check the status of Kubernetes resources:
+     ```bash
+     kubectl get all
+     ```
+   - Ensure that pods are in the `Running` state and services are available.
+
+4. **Access the Application Using Ingress**:
+   - Access the application at:
+     ```
+     http://localhost:8080
```

Comment:
The step for local kubernates deployment should be like this


Step 1: Create the Kubernetes Cluster
`make cluster
`
Step 2: Build and Prepare the Docker Image
If you haven’t already built the Docker image for the application, or if you need to update the image, perform the following steps:
Build the Docker Image Navigate to the directory containing your Dockerfile and run:
`docker build -t promotions:latest .
`

2. Check Existing Docker Images To verify the image was built successfully or check if it already exists, use:
`docker images`
3. Tag and Push the Docker Image :
```
docker tag promotions:latest cluster-registry:5000/promotions:latest
docker push cluster-registry:5000/promotions:latest
```
4. Deploy PostgreSQL and Application Resources
```
kubectl apply -f k8s/postgresql/
kubectl apply -f k8s/
```
5. Verify the Deployment
`kubectl get pods`

6. Access the application at:
     ```
     http://localhost:8080
     ```